src/main/java/com/olgaboiar/guacomole/EchoClient.java

```diff
@@ -0,0 +1,45 @@
+package com.olgaboiar.guacomole;
+
+import java.io.*;
+import java.net.InetSocketAddress;
+import java.net.Socket;
+
+public class EchoClient implements Closeable{
+    String host;
+    int port;
+    Socket socket;
+    int connectionAttempts;
+    int maxConnectionAttempts;
+
+    public EchoClient(String host, int port) {
+        this.host = host;
+        this.port = port;
+        this.maxConnectionAttempts = 20;
+        this.connectionAttempts = 0;
```

Comment:
I would also make these two variables configurable.

Reply:
minor refactoring in https://github.com/olgaboiar/GuacamoleServer/pull/3/commits/aae8861758b3375a3435f7e0d9be1c62448c5d7b